server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java

```diff
@@ -109,10 +109,12 @@ public QualityProfileDto getProfile(DbSession dbSession, QProfileReference ref)
     QualityProfileDto profile;
     if (ref.hasKey()) {
       profile = dbClient.qualityProfileDao().selectByKey(dbSession, ref.getKey());
+      checkFound(profile, "Quality Profile with key '"+ref.getKey()+"' does not exist");
     } else {
       OrganizationDto org = getOrganizationByKey(dbSession, ref.getOrganizationKey().orElse(null));
       profile = dbClient.qualityProfileDao().selectByNameAndLanguage(org, ref.getName(), ref.getLanguage(), dbSession);
+      checkFound(profile, "Quality Profile for language '"+ref.getLanguage()+"' and name '"+ref.getName()+"' does not exist"+ref.getOrganizationKey().map(o ->" in organization '"+o+"'").orElse(""));
```

Comment:
`checkFound` accepts patterns with '%s' (`String.format` style), don't use String concatenation (it has a cost even if check doesn't fail)